Jenkinsfile

```diff
@@ -0,0 +1,319 @@
+        string (defaultValue: 'password', name:'MYSQL_PWD', description:'My Sql user password')
+    }
+    stages {
+        stage('Wait for user to select module to build.') {
+            steps {
+                script {
+                    try {
+                        timeout (time:600, unit:'SECONDS') {
+                            env.userSelModule = input(id: 'userInput', message: 'Please select which module to bulid?',
+                            parameters: [[$class: 'ChoiceParameterDefinition', defaultValue: 'strDef', 
+                               description:'describing choices', name:'nameChoice', choices: "Gateway\nFisUser\nFisAlert\nUI\nAll"]
+                            ])
+                        }
+                    } catch (exception) {
+                      env.userSelModule='All'    
+                    }
+        
+                    println("User selected module " + env.userSelModule);
+                }
+            }
+        }
+        stage('Build maingateway-service') {
+            environment {
+                serviceName = 'maingateway-service'
+            }
+            when {
+                expression {
+                    env.userInput == 'Gateway' || env.userInput == 'All'
+                }
+            }
+            steps {
+                echo "Building.. ${serviceName} "
+                build(env.serviceName)
+
+                echo "Deploying ${serviceName} to ${CICD_PROJECT}"
+                deploy(env.serviceName, params.CICD_PROJECT, params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, params.MYSQL_USER, params.MYSQL_PWD)
+
+            }
+        }
+        stage('Build fisuser-service') {
+            environment { 
+                serviceName = 'fisuser-service'
+            }
+            when {
+                expression {
+                    env.userSelModule == 'FisUser' || env.userSelModule == 'All'
+                }
+            }
+            steps {
+                echo "Building.. ${serviceName} "
+                build(env.serviceName)
+
+                echo "Deploying ${serviceName} to ${CICD_PROJECT}"
+                deploy(env.serviceName, params.CICD_PROJECT, params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, params.MYSQL_USER, params.MYSQL_PWD)
+           }
+        }
+        stage('Build fisalert-service') {
+            environment { 
+                serviceName = 'fisalert-service'
+            }
+            when {
+                expression {
+                    env.userSelModule == 'FisAlert' || env.userSelModule == 'All'
+                }
+            }
+            steps {
+                echo "Building.. ${serviceName} "
+                build(env.serviceName)
+
+                echo "Deploying ${serviceName} to ${CICD_PROJECT}"
+                deploy(env.serviceName, params.CICD_PROJECT, params.OPENSHIFT_HOST, params.OPENSHIFT_TOKEN, params.MYSQL_USER, params.MYSQL_PWD)
```

Comment:
why MYSQL password required to deploy alertService

Reply:
There are here cos originally I want to have a common method for all modules.
We now decided to move to let each individual module to have their own template.  The redundant here can be remove.

Reply:
removed